src/test/java/uk/ac/ebi/eva/contigalias/service/AssemblyAndChromosomeServiceIntegrationTest.java

```diff
@@ -0,0 +1,244 @@
+            }
+        }
+
+        void testAssemblyIdenticalToEntity(Optional<AssemblyEntity> optional) {
+            assertTrue(optional.isPresent());
+            AssemblyEntity assembly = optional.get();
+            assertEquals(assemblyEntity.getName(), assembly.getName());
+            assertEquals(assemblyEntity.getOrganism(), assembly.getOrganism());
+            assertEquals(assemblyEntity.getGenbank(), assembly.getGenbank());
+            assertEquals(assemblyEntity.getRefseq(), assembly.getRefseq());
+            assertEquals(assemblyEntity.getTaxid(), assembly.getTaxid());
+            assertEquals(assemblyEntity.isGenbankRefseqIdentical(), assembly.isGenbankRefseqIdentical());
+        }
+    }
+
+    @Nested
+    class MultipleAssembliesAndChromosomes {
+
+        AssemblyEntity[] assemblyEntities;
+
+        ChromosomeEntity[] chromosomeEntities;
+
+        @BeforeEach
+        void setup() {
+            assemblyEntities = new AssemblyEntity[3];
+            chromosomeEntities = new ChromosomeEntity[4];
+
+            // New assembly
+            assemblyEntities[0] = AssemblyGenerator.generate();
+
+            // Same taxid as previous
+            AssemblyEntity taxidSameAs0th = AssemblyGenerator.generate();
+            taxidSameAs0th.setTaxid(assemblyEntities[0].getTaxid());
+            assemblyEntities[1] = taxidSameAs0th;
+
+            // Totally different
+            assemblyEntities[2] = AssemblyGenerator.generate();
+
+            // New chromosome linked to 0th assembly
+            chromosomeEntities[0] = ChromosomeGenerator.generate(assemblyEntities[0]);
+
+            // Same name as previous and linked to 1st assembly
+            ChromosomeEntity nameSameAs0thAsm1 = ChromosomeGenerator.generate(assemblyEntities[1]);
+            nameSameAs0thAsm1.setName(chromosomeEntities[0].getName());
+            chromosomeEntities[1] = nameSameAs0thAsm1;
+
+            // Same name as previous and linked to 2nd assembly
+            ChromosomeEntity nameSameAs1stAsm1 = ChromosomeGenerator.generate(assemblyEntities[2]);
+            nameSameAs1stAsm1.setName(chromosomeEntities[0].getName());
+            chromosomeEntities[2] = nameSameAs1stAsm1;
+
+            // New chromosome linked to 2nd assembly
+            chromosomeEntities[3] = ChromosomeGenerator.generate(assemblyEntities[2]);
+
+            for (AssemblyEntity assemblyEntity : assemblyEntities) {
+                assemblyService.insertAssembly(assemblyEntity);
+            }
+        }
+
+        @AfterEach
+        void tearDown() {
+            for (AssemblyEntity assemblyEntity : assemblyEntities) {
+                assemblyService.deleteAssembly(assemblyEntity);
+            }
+            assemblyEntities = null;
+            chromosomeEntities = null;
+        }
+
+        /**
+         * In this test two chromosomes have non-distinct names and are linked to different assemblies except that those
```

Comment:
I'm not sure why you say "non-distinct" instead of "the same". Is there any detail you intended to highlight?

Reply:
No specific detail. Just thought it would be too wordy and confusing to explain what is the "same" as to what. The reasoning behind these tests being separate is a bit confusing at first but I promise they test different cases.

Reply:
yeah I understand the tests themselves, I was talking about the comments only.